### PR TITLE
[Text fields] Trailing view support

### PR DESCRIPTION
**Pull request by willlarche**
Dependent on #1743

### CODE REVIEW

File: components/TextFields/src/MDCMultilineTextField.m [hunk 1]
```diff
@@ -46,6 +46,9 @@ @interface MDCMultilineTextField () {
 @property(nonatomic, strong) NSLayoutConstraint *textViewMinHeight;
 @property(nonatomic, strong) NSLayoutConstraint *textViewTop;
 @property(nonatomic, strong) NSLayoutConstraint *textViewTrailing;
+@property(nonatomic, strong) NSLayoutConstraint *textViewTrailingTrailingViewLeading;
```
Comment (ianegordon) [inline components/TextFields/src/MDCMultilineTextField.m]: TrailingTrailingViewLeading?
Would you add a comment to explain.

File: components/TextFields/src/MDCMultilineTextField.m [hunk 2]
```diff
@@ -387,6 +394,72 @@ - (CGFloat)estimatedTextViewLineHeight {
...
 }
 
+#pragma mark - Trailing View Implementation
+
+- (void)updateTrailingViewLayout {
+  if (!self.trailingView) {
+    return;
+  }
+
+  self.trailingView.translatesAutoresizingMaskIntoConstraints = NO;
```
Comment (ianegordon) [inline components/TextFields/src/MDCMultilineTextField.m:407]: I would only expect to set this once, when trailingView is created or added.
There may be a deeper reason I'm not seeing.  If so, perhaps a comment.

File: components/TextFields/src/MDCMultilineTextField.m [hunk 3]
```diff
@@ -387,6 +394,72 @@ - (CGFloat)estimatedTextViewLineHeight {
...
+
+  if (![self.trailingView isDescendantOfView:self]) {
+    [self addSubview:self.trailingView];
+  }
+
+  if (!self.trailingViewTrailing) {
```
Comment (ianegordon) [inline components/TextFields/src/MDCMultilineTextField.m:413]: Once created, will self.trailingViewTrailing ever be nil again?
dispatch_once ?
(dispatch_once may be less clear)
Perhaps a comment?
// Create trailing view trailing constraint if it doesn't exist yet.
If this is only created once, will it break if we change self.trailingView?

**Review by ianegordon — CHANGES_REQUESTED** [review]

**Review by ianegordon — APPROVED** [review]

**Comment by ianegordon** [comment]
Discussed offline: The constraints will get destroyed if the view is changed.  This is why they are recreated in-line.
Please merge when Travis is happy!

File: components/TextFields/src/MDCMultilineTextField.m (hunk 1 above)
Comment (willlarche) [reply components/TextFields/src/MDCMultilineTextField.m]: Done!

**Pull request merged**